analyzer/decrypt_crypto.cc

```diff
@@ -83,45 +83,141 @@ const size_t AEAD_TAG_LENGTH = 16;
 const size_t MAXIMUM_PACKET_LENGTH = 1500;
 const size_t MAXIMUM_PACKET_NUMBER_LENGTH = 4;
 
+EVP_CIPHER_CTX* get_aes_128_ecb()
+	{
+	static EVP_CIPHER_CTX* ctx = nullptr;
+	if ( ! ctx )
+		{
+		ctx = EVP_CIPHER_CTX_new();
```

Comment:
Just as a note, in multi-threaded code this would not be safe since `ctx` might be set once you reach this assignment. You'd either need to use double-checked locking, or use a `static` stack variable which is guaranteed by the language to be initialized exactly once on first use, e.g.,

```cpp
EVP_CIPHER_CTX* get_aes_128_ecb()
	{
	static struct Ctx
		{
		Ctx() : val(EVP_CIPHER_CTX_new())
			{
			EVP_CipherInit_ex(val, EVP_aes_128_ecb(), NULL, NULL, NULL, 1);
			}
		EVP_CIPHER_CTX* val;
		} ctx;

	return ctx.val;
	}
```